### PR TITLE
Adds running over people with mechs

### PR DESCRIPTION
**Pull request by Alberyk**
Walking over someone lying on the floor with mechs will now hurt them. Damage is based on the legs.
Also, removed some baylore references to the mech parts, and made so that mechs can push mobs around.

### CODE REVIEW

File: code/modules/heavy_vehicle/mech_interaction.dm [hunk 1]
```diff
@@ -462,4 +462,31 @@
...
+		return
+	if(!isliving(H))
+		return
+
+	if(legs)
+		if(ishuman(H))
+			var/mob/living/carbon/human/D = H
+			if(D.lying)
+				D.attack_log += "\[[time_stamp()]\]<font color='orange'> Was trampled by [src]</font>"
+				attack_log += text("\[[time_stamp()]\] <font color='red'>trampled [D.name] ([D.ckey]) with the [src].</font>")
```
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/mech_interaction.dm]:
```suggestion
				attack_log += text("\[[time_stamp()]\] <font color='red'>trampled [D.name] ([D.ckey]) with \the [src].</font>")
```

File: code/modules/heavy_vehicle/mecha.dm [hunk 2]
```diff
@@ -7,6 +7,7 @@
 	status_flags = PASSEMOTES
 	a_intent = I_HURT
 	mob_size = MOB_LARGE
+	mob_push_flags = ALLMOBS
```
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/mecha.dm]: Undocumented in the changelog?

**Comment by KasparoVy** [comment]
Wicked!

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -817,12 +817,16 @@ default behaviour is:
...
 /proc/is_valid_for_devour(var/mob/living/test, var/eat_types)
 	//eat_types must contain all types that the mob has. For example we need both humanoid and synthetic to eat an IPC.
 	var/test_types = test.find_type()
 	. = (eat_types & test_types) == test_types
- 
+
+/mob/living/Crossed(var/atom/movable/AM)
+	if(istype(AM, /mob/living/heavy_vehicle))
+		var/mob/living/heavy_vehicle/MB = AM
+		MB.trample(src)
```
Comment (skull132) [inline code/modules/mob/living/living.dm]:
```suggestion
		MB.trample(src)
	..()
```

**Pull request merged**